synapse_invite_checker/config.py

```
@@ -17,6 +17,12 @@
 from synapse_invite_checker.types import TimType
 
 
+@dataclass
+class InsuredOnlyRoomScanConfig:
+    epa_room_grace_period_ms: int = 0
```

Not sure if the epa makes that much sense in the name, but I guess with the current spec it does.
Also maybe this shouldn't default to a grace period of 0 and should default to enabled?

---

It should only default to enabled on epa servers, I think. Provided that Synapse's `forget_rooms_on_leave` and `forgotten_room_retention_period` are set, it serves no purpose to run on pro servers?

But that did make me question how the defaults are implemented, I'll go take another look at that

---

Focusing on this the grace period defaulting to `0` here actually seems ok. And here's why I think that:
* I did change out how the config is parsed in `parse_config()` so that it always has at the minimum an empty `Dict` to look at(and not None anymore)
* Since that logic no longer hides it all behind checking for the None, any defaults defined in `parse_config()` should now work unilaterally
* In the chance of a `ConfigError` raising(something I just added), all of this is moot and the `0` would be ignored and the room scan would be disabled anyway because of the `enabled` also being False

Does that make sense?

---

The rest of the adjustments here were done in [916b94b](https://github.com/famedly/synapse-invite-checker/pull/43/commits/916b94b162374ebd31e7789640b8193042581beb)

---

Yeah, that's fine. It just seemed weird, because I read the diff from the top.

And yes, it only makes sense to run this on pro servers, I just think the "epa" makes little sense to include in the config key, since the whole check is epa only. I think the check should just silently do nothing when enabled on pro servers.

---

See what you think about it now in [d515637](https://github.com/famedly/synapse-invite-checker/pull/43/commits/d515637e7340e54713809a5899f5209512773bb9)